camelot/view/crud_action.py

```diff
@@ -44,20 +48,67 @@ def strip_data_from_object( obj, columns ):
     return row_data
 
 
+non_serializable_roles = (
+    Qt.ItemDataRole.DisplayRole, Qt.ItemDataRole.AccessibleDescriptionRole
+)
+
+@dataclass
+class DataCell(DataclassSerializable):
+
+    row: int = -1
+    column: int = -1
+    flags: int = Qt.ItemFlag.ItemIsEnabled | Qt.ItemFlag.ItemIsDropEnabled | Qt.ItemFlag.ItemIsDragEnabled | Qt.ItemFlag.ItemIsEditable | Qt.ItemFlag.ItemIsSelectable
+
+    roles: Dict[int, Any] = field(default_factory=dict)
+
+    # used in camelot tests
+    def get_standard_item(self):
+        item = QtGui.QStandardItem()
+        item.setFlags(self.flags)
+        for role, value in self.roles.items():
+            item.setData(value, role)
+        return item
+
+@dataclass
+class DataRowHeader(DataclassSerializable):
+
+    row: int = -1
+    tool_tip: Optional[str] = None
+    icon_name: Optional[str] = None
+    object: int = 0
+    verbose_identifier: str = ''
+    valid: bool = True
+    message: str = ''
+    decoration: Optional[Icon] = None
+    display: Optional[str] = None
+
+@dataclass
+class DataUpdate(DataclassSerializable):
+
+    changed_ranges: InitVar
+
+    header_items: List[DataRowHeader] = field(default_factory=list)
+    cells: List[DataCell] = field(default_factory=list)
+
+    def __post_init__(self, changed_ranges):
+        for row, header_item, items in changed_ranges:
+            self.header_items.append(header_item)
+            self.cells.extend(items)
+
+
+invalid_item = DataCell()
```

Comment:
TBD : het is nu wel verdacht dat er 2 definities van invalid_item zijn, nl eentje server en eentje client side ...